test/datasets/test_cnndm.py

```diff
@@ -0,0 +1,97 @@
+import hashlib
+import os
+import tarfile
+from collections import defaultdict
+from unittest.mock import patch
+
+from parameterized import parameterized
+from torchtext.datasets.cnndm import CNNDM
+
+from ..common.case_utils import TempDirMixin, zip_equal, get_random_unicode
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+
+    base_dir = os.path.join(root_dir, "CNNDM")
+    temp_dataset_dir = os.path.join(base_dir, "temp_dataset_dir")
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+    seed = 1
+    mocked_data = defaultdict(list)
+
+    for source in ["cnn", "dailymail"]:
+        source_dir = os.path.join(temp_dataset_dir, source, "stories")
+        os.makedirs(source_dir, exist_ok=True)
+
+        for split in ["train", "val", "test"]:
+            url = source + "_" + split
+            h = hashlib.sha1()
+            h.update(url.encode())
+            filename = h.hexdigest() + ".story"
+            txt_file = os.path.join(source_dir, filename)
+
+            with open(txt_file, "w", encoding=("utf-8")) as f:
+                article = get_random_unicode(seed) + "."
+                abstract = get_random_unicode(seed + 1) + "."
+                dataset_line = (article, abstract)
+                f.writelines([article, "\n@highlight\n", abstract])
+                # append line to correct dataset split
+                mocked_data[split].append(dataset_line)
```

Comment:
nit: I wonder if we should create multiple URLs per story type (CNN, dailymail) like the real datasets? In general I think it's fine what we did here.